Source/Core/Core/Boot/Boot.cpp

```diff
@@ -221,9 +222,8 @@ BootParameters::IPL::IPL(DiscIO::Region region_, Disc&& disc_) : IPL(region_)
 static const DiscIO::Volume* SetDisc(std::unique_ptr<DiscIO::Volume> volume,
                                      std::vector<std::string> auto_disc_change_paths = {})
 {
-  const DiscIO::Volume* pointer = volume.get();
   DVDInterface::SetDisc(std::move(volume), auto_disc_change_paths);
-  return pointer;
+  return DVDThread::GetDiscVolume();
```

Comment:
This is a really ugly situation here. It might actually be cleaner to return to the state before PR #4241

Right now I'm just making it clearer that we are grabbing a pointer internal state of DVDThread (and grabbing it after DVDThread has applyed the patch overlay).

I think the cleaner solution might be to split the EmulatedBS2 into two parts. Part one which sets up the memory/state based on the volume which runs before passing the volume to DVDThread. Part 2 will just load and run apploader, directly calling DVDThread to process read requests (adding a new `DVDThread::HLE_Read()` which is only intended to be used during boot)

But I'm not sure, I welcome feedback on this issue.